src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -79,6 +81,36 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code String speciality} into an {@code Speciality}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code speciality} is invalid.
+     */
+    public static Specialty parseSpeciality(String speciality) throws ParseException {
+        requireNonNull(speciality);
+        String trimmedSpeciality = speciality.trim();
+        if (!Specialty.isValidSpecialty(trimmedSpeciality)) {
+            throw new ParseException(Specialty.MESSAGE_CONSTRAINTS);
+        }
+        return new Specialty(trimmedSpeciality);
+    }
+
+    /**
+     * Parses a {@code String yoe} into an {@code Yoe}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code yoe} is invalid.
+     */
+    public static Yoe parseYoe(String yoe) throws ParseException {
+        requireNonNull(yoe);
+        String trimmedYoe = yoe.trim();
+        if (!Specialty.isValidSpecialty(trimmedYoe)) {
```

Comment:
This should be `(!Yoe.isValidYoe(trimmedYoe)`

Reply:
Resolved